py3status/modules/conky.py

```diff
@@ -0,0 +1,448 @@
+    {'full_text': '5.82%', 'color': '#90ee90'},
+]
+
+network
+[
+    {'full_text': 'Speed '},
+    {'color': '#ff6699', 'full_text': '15B/84B '},
+    {'full_text': 'Total '},
+    {'color': '#ff6699', 'full_text': '249MiB/4.27GiB'},
+]
+
+memory
+[
+    {'full_text': 'Memory '},
+    {'full_text': '2.68G/7.72G ', 'color': '#87cefa'},
+    {'full_text': '34% [###.......]', 'color': '#8ae234'}
+]
+
+swap
+[
+    {'full_text': 'Swap '},
+    {'full_text': '4.5MiB/7.72GiB ', 'color': '#f08080'},
+    {'full_text': '0% [..........]', 'color': '#a9a9a9'}
+]
+
+disk
+[
+    {'full_text': 'Home '},
+    {'full_text': '167G/431G ', 'color': '#ee82ee'},
+    {'full_text': '38% [####......]', 'color': '#8ae234'},
+]
+
+nvidia
+[
+    {'full_text': 'GPU Temp '}, {'full_text': '64 ', 'color': '#adff2f'},
+    {'full_text': 'GPU Freq '}, {'full_text': '460 ', 'color': '#adff2f'},
+    {'full_text': 'Mem Freq '}, {'full_text': '695', 'color': '#adff2f'},
+]
+
+nvidia
+[
+    {'full_text': 'MTR Freq '}, {'full_text': '1390 ', 'color': '#adff2f'},
+    {'full_text': 'Perf '}, {'full_text': '1 ', 'color': '#adff2f'},
+    {'full_text': 'Mem Perc '}, {'full_text': '61', 'color': '#adff2f'},
+]
+
+bar
+[
+    {'full_text': '#.... ', 'color': '#ffffff'},
+    {'full_text': '##... ', 'color': '#00FF00'},
+    {'full_text': '###.. ', 'color': '#FFA500'},
+    {'full_text': '####. ', 'color': '#FFFF00'},
+    {'full_text': '#####', 'color': '#FF0000'},
+]
+"""
+
+from subprocess import Popen, PIPE, STDOUT
+from threading import Thread
+from tempfile import NamedTemporaryFile
+from json import dumps
+from os import remove as os_remove
+
+STRING_NOT_INSTALLED = "not installed"
+STRING_MISSING_FORMAT = "missing format"
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    config = {}
+    format = None
+    thresholds = []
+
+    def post_config_hook(self):
+        if not self.py3.check_commands("conky"):
+            raise Exception(STRING_NOT_INSTALLED)
+        elif not self.format:
+            raise Exception(STRING_MISSING_FORMAT)
+
+        # placeholders
+        self.delimiter = getattr(self, "delimiter", ".")
+        placeholders = self.py3.get_placeholders_list(self.format)
+        _placeholders = [x.replace(self.delimiter, " ") for x in placeholders]
+        colors = self.py3.get_color_names_list(self.format)
+        _colors = []
+        for color in colors:
+            if not getattr(self, "color_{}".format(color), None):
+                _colors.append(color.replace(self.delimiter, " "))
+        self.placeholders = placeholders + colors
+        conky_placeholders = _placeholders + _colors
+
+        # init
+        self.cache_names = {}
+        self.thresholds_init = colors
+        self.config.update({"out_to_x": False, "out_to_console": True})
+        self.separator = "|SEPARATOR|"  # must be upper
+
+        # make an output. TODO: make a better lua->python conversion
```

Comment:
Is the current approach known to fail in some cases? Do you want to tackle this TODO before merging this module? If not, I suggest removing the TODO.

Reply:
Yes on failing in some cases. No on tackling this because I already tried and I still hadn't found a solution. Still an TODO issue.

Reply:
Maybe open an issue for this, instead of having TODO in the code? I don't know how often we search for TODOs in the code, could easily forget about this.

Reply:
TODO that we aren't going to address makes even less sense to me, let's just remove it and wait for a bug report.